Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -153,3 +160,222 @@ theorem LSeries_eq_mul_integral_of_nonneg (f : ℕ → ℝ) {r : ℝ} (hr : 0 
+        tendsto_nat_floor_atTop
+    simpa using hlim
+  refine h₁.integrableOn_of_isBigO_atTop (g := fun t ↦ t ^(-s)) ?_ ?_
+  · refine IsBigO.mul_atTop_rpow_of_isBigO_rpow 1 (-s - 1) _ h₂ ?_ (by linarith)
+    exact (norm_ofReal_cpow_eventually_eq_atTop _).isBigO.of_norm_left
+  · rwa [integrableAtFilter_rpow_atTop_iff, neg_lt_neg_iff]
+
+private theorem lemma₂ {s T ε : ℝ} {S : ℝ → ℂ} (hs : 1 < s) (hT : 0 < T)
+    (hS₁ : LocallyIntegrableOn (fun t ↦ S t) (Set.Ici 1)) (hS₂ : ∀ t ≥ T, ‖S t‖ ≤ ε * t) :
+    IntegrableOn (fun t : ℝ ↦ ‖S t‖ * (t ^ (-s - 1))) (Set.Ici 1) := by
+  have h : LocallyIntegrableOn (fun t : ℝ ↦ ‖S t‖ * (t ^ (-s - 1))) (Set.Ici 1) := by
+    refine hS₁.norm.mul_continuousOn ?_ isLocallyClosed_Ici
+    exact fun t ht ↦ (Real.continuousAt_rpow_const _ _
+      <| Or.inl (zero_lt_one.trans_le ht).ne').continuousWithinAt
+  refine h.integrableOn_of_isBigO_atTop (g := fun t ↦ t ^(-s)) (isBigO_iff.mpr ⟨ε, ?_⟩) ?_
+  · filter_upwards [eventually_ge_atTop T] with t ht
+    have ht' : 0 < t := hT.trans_le ht
+    rw [norm_mul, norm_norm, show t ^ (-s) = t * t ^ (-s - 1) by
+        rw [Real.rpow_sub ht', Real.rpow_one, mul_div_cancel₀ _ ht'.ne'],
+      norm_mul, Real.norm_of_nonneg ht'.le, ← mul_assoc]
+    exact mul_le_mul_of_nonneg_right (hS₂ t ht) (norm_nonneg _)
+  · exact integrableAtFilter_rpow_atTop_iff.mpr <| neg_lt_neg_iff.mpr hs
+
+end lemmas
+
+section proof
+
+include hlim in
+private theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₁ {ε : ℝ} (hε : ε > 0) :
+    ∀ᶠ t : ℝ in atTop, ‖(∑ k ∈ Icc 1 ⌊t⌋₊, f k) - l * t‖ < ε * t := by
+  have h_lim' : Tendsto (fun t : ℝ ↦ (∑ k ∈ Icc 1 ⌊t⌋₊, f k : ℂ) / t) atTop (𝓝 l) := by
+    refine (mul_one l ▸ ofReal_one ▸ ((hlim.comp tendsto_nat_floor_atTop).mul <|
+      tendsto_ofReal_iff.mpr <| tendsto_nat_floor_div_atTop)).congr' ?_
+    filter_upwards [eventually_ge_atTop 1] with t ht
+    rw [Function.comp_def, ofReal_div, ofReal_natCast, div_mul_div_cancel₀ (by
+      rwa [Nat.cast_ne_zero, ne_eq, Nat.floor_eq_zero, not_lt])]
+  rw [Metric.tendsto_nhds] at h_lim'
+  filter_upwards [eventually_gt_atTop 0, h_lim' ε hε] with t ht₁ ht₂
+  rwa [dist_eq_norm, div_sub' _ _ _ (ne_zero_of_re_pos ht₁), norm_div, norm_real,
+    Real.norm_of_nonneg ht₁.le, mul_comm, div_lt_iff₀ ht₁] at ht₂
+
+private theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₂ {s T ε : ℝ} {S : ℝ → ℂ}
+    (hS : LocallyIntegrableOn (fun t ↦ S t - l * t) (Set.Ici 1)) (hε : 0 < ε)
+    (hs : 1 < s) (hT₁ : 1 ≤ T) (hT : ∀ t ≥ T, ‖S t - l * t‖ ≤ ε * t) :
+    (s - 1) * ∫ (t : ℝ) in Set.Ioi T, ‖S t - l * t‖ * t ^ (-s - 1) ≤ ε := by
+  have hT₀ : 0 < T := zero_lt_one.trans_le hT₁
+  have h {t : ℝ} (ht : 0 < t) : t ^ (-s) = t * t ^ (-s - 1) := by
+    rw [Real.rpow_sub ht, Real.rpow_one, mul_div_cancel₀ _ ht.ne']
+  calc
+    _ ≤ (s - 1) * ∫ (t : ℝ) in Set.Ioi T, ε * t ^ (-s) := ?_
+    _ ≤ ε * ((s - 1) * ∫ (t : ℝ) in Set.Ioi 1, t ^ (-s)) := ?_
+    _ = ε := ?_
```

Comment:
I would personally find it clearer if the proofs of the calc steps were interleaved between the calc lines rather than being stored up and done at the end.